metaseq/cli/train.py

```diff
@@ -495,6 +495,57 @@ def post_checkpoint_callback(cfg, filename):
                 ),
             )
             os.remove(filename)
+
+            # Run evals after all checkpoint parts have been uploaded
+            checkpoint_id = checkpoint_dir.split("_")[-1]
+            if (
+                cfg.checkpoint.cloud_eval_script_path is not None
+                and distributed_utils.get_global_rank() == 0
+                and checkpoint_id.isdigit()
+                and cfg.checkpoint.cloud_eval_frequency > 0
+                and (int(checkpoint_id) % cfg.checkpoint.cloud_eval_frequency == 0)
+            ):
+                for retry in range(cfg.checkpoint.cloud_eval_num_retries):
+                    time.sleep(cfg.checkpoint.cloud_eval_retry_wait_minutes * 60)
+
+                    current_checkpoint_path = os.path.join(
+                        destination_checkpoints_dir, checkpoint_dir
+                    )
+                    num_files = os.listdir(current_checkpoint_path)
+                    finished_checkpoint_parts = len(
+                        [f for f in num_files if not f.startswith("_")]
+                    )
+                    if (
+                        finished_checkpoint_parts
+                        == cfg.distributed_training.distributed_world_size
+                    ):
+                        logger.info(
+                            f"All checkpoint parts for {checkpoint_dir} are in NFS, will now start to run evals"
+                        )
+                        script_dir = os.path.join(
```

Comment:
I don't feel particularly strong either way, but @stephenroller had suggested that we use a python function instead, see comments in https://github.com/facebookresearch/metaseq/pull/411/files. Might be easier for version control?

Reply:
yes plz

Reply:
Do you mean that instead of running a bash script, we run a python function? Mhm, but we are currently using Punit's bash script for these evals. So we might need to translate that into python first  and validate it again. Not sure that is worth it?

Reply:
Yea if we are already using the bash script everywhere then I agree to keep it this way, at least for now